src/base/bittorrent/session.cpp

```diff
@@ -3933,6 +3944,78 @@ void Session::handleTorrentTrackerError(TorrentHandle *const torrent, const QStr
     emit trackerError(torrent, trackerUrl);
 }
 
+bool Session::addMoveTorrentStorageJob(TorrentHandle *torrent, const QString &newPath, const MoveStorageMode mode)
+{
+    Q_ASSERT(torrent);
+
+    if (m_moveStorageQueue.size() > 1) {
+        const auto iter = std::find_if(m_moveStorageQueue.begin() + 1, m_moveStorageQueue.end()
+                                 , [torrent](const MoveStorageJob &job)
+        {
+            return job.torrent == torrent;
+        });
+
+        if (iter != m_moveStorageQueue.end()) {
+            // remove existing inactive job
+            m_moveStorageQueue.erase(iter);
+        }
+    }
+
+    QString currentLocation = QString::fromStdString(
+                torrent->nativeHandle().status(lt::torrent_handle::query_save_path).save_path);
+    if (!m_moveStorageQueue.isEmpty() && (m_moveStorageQueue.first().torrent == torrent)) {
+        // if there is active job for this torrent consider its target path as current location
+        // of this torrent to prevent creating meaningless job that will do nothing
+        currentLocation = m_moveStorageQueue.first().path;
+    }
+
+    if (QDir {currentLocation} == QDir {newPath})
+        return false;
+
+    const MoveStorageJob moveStorageJob {torrent, newPath, mode};
```

Comment:
Damn it! I additionally removed appending new job to the queue when I had applying proposed fixes... so #12185 is here!
As I said there I'm away from any computer so I can't fix it. Can someone help?

Reply:
>As I said there I'm away from any computer so I can't fix it. Can someone help?

@thalieht, ping!

Reply:
@glassez I looked at this for a while but I have no idea how to fix it, sorry :(

Reply:
You just need to append `moveStorageJob` to `m_moveStorageQueue` in the following line.

Reply:
#12188

Reply:
I didn't either but i looked at the previous forced pushed commits.

Reply:
@thalieht interesting, I had no idea you could see the previous force-pushed commits. I thought once you forced-pushed, the previous commits were overwritten.

Reply:
They are overwritten but for some reason github saves them. I can't diff them or i don't know how so this is the only time i've found this useful because i knew exactly where to look.

Reply:
Whoa they can be diffed by clicking on "force-pushed". Just found out.